components/scream/src/physics/spa/spa_functions_impl.hpp

```diff
@@ -0,0 +1,285 @@
+      const auto& aer_tau_sw_src_sub         = ekat::subview(aer_tau_sw_src, i, n);
+
+      /* Now loop over fastest index, the number of vertical packs */
+      Kokkos::parallel_for(
+        Kokkos::ThreadVectorRange(team, nk_pack), [&] (Int k) {
+        aer_g_sw_src_sub(k)   = linear_interp(aer_g_sw_beg_sub(k),   aer_g_sw_end_sub(k),   t_norm);
+        aer_ssa_sw_src_sub(k) = linear_interp(aer_ssa_sw_beg_sub(k), aer_ssa_sw_end_sub(k), t_norm);
+        aer_tau_sw_src_sub(k) = linear_interp(aer_tau_sw_beg_sub(k), aer_tau_sw_end_sub(k), t_norm);
+      });
+    });
+    team.team_barrier();
+    }
+    {
+    /* Loop over all LW variables with nlwbands */
+    Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, nlwbands), [&] (int n) {
+      const auto& aer_tau_lw_beg_sub         = ekat::subview(data_beg.AER_TAU_LW, i, n);
+      const auto& aer_tau_lw_end_sub         = ekat::subview(data_end.AER_TAU_LW, i, n);
+      const auto& aer_tau_lw_src_sub         = ekat::subview(aer_tau_lw_src, i, n);
+
+      /* Now loop over fastest index, the number of vertical packs */
+      Kokkos::parallel_for(
+        Kokkos::ThreadVectorRange(team, nk_pack), [&] (Int k) {
+        aer_tau_lw_src_sub(k) = linear_interp(aer_tau_lw_beg_sub(k), aer_tau_lw_end_sub(k), t_norm);
+      });
+    });
+    team.team_barrier();
+    }
+  });
+  Kokkos::fence();
+
+  // Third Step: Vertical interpolation, project the SPA data onto the pressure profile for this simulation.
+  // This is done using the EKAT linear interpolation routine, see /externals/ekat/util/ekat_lin_interp.hpp
+  // for more details. 
+  using LIV = ekat::LinInterp<Real,Spack::n>;
+  Real minthreshold = 0.0;  // Hard-code a minimum value for aerosol concentration to zero.
+
+  LIV VertInterp(ncols_atm,pressure_state.nlevs,nlevs_atm,minthreshold);
+  /* Parallel loop strategy:
+   * 1. Loop over all simulation columns (i index)
+   * 2. Where applicable, loop over all aerosol bands (n index)
+   */ 
+  Kokkos::parallel_for("vertical-interp-spa",
+    VertInterp.m_policy,
+    KOKKOS_LAMBDA(typename LIV::MemberType const& team) {
+    const int i = team.league_rank();
+    /* Setup the linear interpolater for this column. */
+    VertInterp.setup(team,
+                     ekat::subview(p_src,i),
+                     ekat::subview(pressure_state.pmid,i));
+    team.team_barrier();
+    /* Conduct vertical interpolation for the 2D variable CCN3 */
+    VertInterp.lin_interp(team,
+                          ekat::subview(p_src,i),
+                          ekat::subview(pressure_state.pmid,i),
+                          ekat::subview(ccn3_src,i),
+                          ekat::subview(data_out.CCN3,i));
+    /* Conduct vertical interpolation for the LW banded data - nlwbands (n index) */
+    Kokkos::parallel_for(
+      Kokkos::TeamThreadRange(team, nlwbands), [&] (int n) {
+      VertInterp.lin_interp(team,
```

Comment:
You'll need to set up a ThreadVectorRange here:
```
const auto& tvr = Kokkos::ThreadVectorRange(team, VertInterp.km2_pack());
```
... and pass it as the second arg to lin_interp.